src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValue.cs

```diff
@@ -58,5 +75,19 @@ public string ToExpression()
 
             return sb.ToString();
         }
+    }
+
+    /// <summary>
+    /// Indicates that a FormulaValue should be copied before being mutated, for Copy on Write semantics.
+    /// </summary>
+    internal interface IMutationCopy

```

Comment:
If DV needs to implement, this would need to be public.

Reply:
Agreed.  We'll look at this again after we get through the Dataverse issues.